verbose: 0      # Residual print = 1, DistanceToGoal check = 2, Collision print = 3, Compute time print = 4, Scaled Thurst print = 5
                              
num_drone: 4 # ignored if read_config is false
num_obs: 56    # ignored if read_config is false

# @@ For axis_wise set true otherwise quadratic
axis_wise: false     
jerk_snap_constraints: false
admm : false
mode : 2 #mode 1：AMswarm 2：Priority-based 3：Ours
# @@ CBF 
gamma: 1

# @@ Planning Parameters
world:      2     
num:        30
t_plan:     3
num_up:     300
dist_stop:  0.2
free_space: false
kappa:      5
free_dynamic: true #new add

# @@ Optimizer Parameters
max_iter: 3000                  
max_time: 20
thresold: 0.01

order_smoothness:  2
weight_goal:       10000        
weight_smoothness: 300
delta_aggressive:  1

delta_static_obs: 1.3
delta_dynamic_obs: 1.3
delta_drone:      1.3
delta_vel:        1.3
delta_acc:        1.3
delta_jerk:       1.3
delta_snap:       1.3
delta_ineq:       1.3

rho_static_obs_max: 50000000
rho_dynamic_obs_max: 50000000
rho_drone_max:      50000000
rho_vel_max:        50000000
rho_acc_max:        50000000
rho_jerk_max:       50000000
rho_snap_max:       50000000
rho_ineq_max:       50000000

# true kinematic values
vel_max: 1.5
jerk_max: 8.0
snap_max: 8.0

# scaled thrust values 
use_thrust_values: true
gravity: 9.8
f_max: 1.5    # 1.5g
f_min: 0.3    # 0.3g


acc_max: 2.8

prox_agent: 0.2
prox_obs: 0.2

buffer: 0.05

# @@ Crazyflie Configuration 
a_drone: 0.0675
b_drone: 0.0675
c_drone: 0.200


# @@ Room limits
x_lim: [-2.0, 2.0]
y_lim: [-2.0, 2.0]
z_lim: [+0.2, 2.2]



# @@ Obstacle Configuration 


pos_static_obs: [
                
                
                 [-0.9 ,0.4, 0],
                 [0.9,0.4, 0],
                [0.6 ,0.4, 0],
                [0.3 ,0.4, 0],
               [0,0.4,0],
           [0.3 ,-0.4, 0],
                [0.6 ,-0.4, 0],
               [0,-0.4,0],
                [-0.3,0.4, 0],
                [-0.6,0.4, 0],
         
                [-0.6,-0.4, 0],
                 [-0.3,-0.4, 0],
                  [-0.9 ,-0.4, 0],
                   [0.9 ,-0.4, 0],
                   

                   [-0.9 ,0.8, 0],
                 [0.9,0.8, 0],
                [0.6 ,0.8, 0],
                [0.3 ,0.8, 0],
               [0,0.8,0],
           [0.3 ,-0.8, 0],
                [0.6 ,-0.8, 0],
               [0,-0.8,0],
                [-0.3,0.8, 0],
                [-0.6,0.8, 0],
         
                [-0.6,-0.8, 0],
                 [-0.3,-0.8, 0],
                  [-0.9 ,-0.8, 0],
                   [0.9 ,-0.8, 0],

                   [-0.9 ,1.2, 0],
                 [0.9,1.2, 0],
                [0.6 ,1.2, 0],
                [0.3 ,1.2, 0],
               [0,1.2,0],
           [0.3 ,-1.2, 0],
                [0.6 ,-1.2, 0],
               [0,-1.2,0],
                [-0.3,1.2, 0],
                [-0.6,1.2, 0],
         
                [-0.6,-1.2, 0],
                 [-0.3,-1.2, 0],
                  [-0.9 ,-1.2, 0],
                   [0.9 ,-1.2, 0],
               
                [-0.9 ,1.6, 0],
                 [0.9,1.6, 0],
                [0.6 ,1.6, 0],
                [0.3 ,1.6, 0],
               [0,1.6,0],
           [0.3 ,-1.6, 0],
                [0.6 ,-1.6, 0],
               [0,-1.6,0],
                [-0.3,1.6, 0],
                [-0.6,1.6, 0],
         
                [-0.6,-1.6, 0],
                 [-0.3,-1.6, 0],
                  [-0.9 ,-1.6, 0],
                   [0.9 ,-1.6, 0],
                
              
                ]

dim_static_obs: [         
               [+0.27, +0.27, 10.0],       
            [+0.27, +0.27, 10.0],   
                  [+0.27, +0.27, 10.0],   
              [+0.27, +0.27, 10.0],             
               [+0.27, +0.27, 10.0],       
            [+0.27, +0.27, 10.0],
                [+0.27, +0.27, 10.0],   
              [+0.27, +0.27, 10.0],             
               [+0.27, +0.27, 10.0],       
            [+0.27, +0.27, 10.0],
            [+0.27, +0.27, 10.0],   
              [+0.27, +0.27, 10.0],             
               [+0.27, +0.27, 10.0],       
            [+0.27, +0.27, 10.0],
            [+0.27, +0.27, 10.0],       
            [+0.27, +0.27, 10.0],   
                  [+0.27, +0.27, 10.0],   
              [+0.27, +0.27, 10.0],             
               [+0.27, +0.27, 10.0],       
            [+0.27, +0.27, 10.0],
                [+0.27, +0.27, 10.0],   
              [+0.27, +0.27, 10.0],             
               [+0.27, +0.27, 10.0],       
            [+0.27, +0.27, 10.0],
            [+0.27, +0.27, 10.0],   
              [+0.27, +0.27, 10.0],             
               [+0.27, +0.27, 10.0],       
            [+0.27, +0.27, 10.0],
            [+0.27, +0.27, 10.0],       
            [+0.27, +0.27, 10.0],   
                  [+0.27, +0.27, 10.0],   
              [+0.27, +0.27, 10.0],             
               [+0.27, +0.27, 10.0],       
            [+0.27, +0.27, 10.0],
                [+0.27, +0.27, 10.0],   
              [+0.27, +0.27, 10.0],             
               [+0.27, +0.27, 10.0],       
            [+0.27, +0.27, 10.0],
            [+0.27, +0.27, 10.0],   
              [+0.27, +0.27, 10.0],             
               [+0.27, +0.27, 10.0],
               [+0.27, +0.27, 10.0],
               [+0.27, +0.27, 10.0],       
            [+0.27, +0.27, 10.0],   
                  [+0.27, +0.27, 10.0],   
              [+0.27, +0.27, 10.0],             
               [+0.27, +0.27, 10.0],       
            [+0.27, +0.27, 10.0],
                [+0.27, +0.27, 10.0],   
              [+0.27, +0.27, 10.0],             
               [+0.27, +0.27, 10.0],       
            [+0.27, +0.27, 10.0],
            [+0.27, +0.27, 10.0],   
              [+0.27, +0.27, 10.0],             
               [+0.27, +0.27, 10.0],
               [+0.27, +0.27, 10.0],
                     
                       
               

                          
               ]
                
pos_dynamic_obs: [[2.5, 0, 0]]
dim_dynamic_obs: [[+0.10, +0.10, 10.0]]          
dynamic_obs_v:  [[-0.09, 0, 0]]
init_drone: [[0.8,0,0.5],
              [-0.8,0,0.5],[1.8,0,0.5],[0,0,0.5],[-1.8,0.8,0.5],[1.8,0.5,0.5],[-1.8,-0.5,0.5],[1.8,-0.5,0.5]]
#-0.05 ->0
goal_drone:  [[-1.8,0,0.5],
               [1.5,0,0.5],
               [-1.8,0.5,0.5] ,
               [-1.8,-0.5,0.5],[1.8,0,0.5],[-1.5,-0.05,0.5],[1.3,0,0.5],[-1.5,-0.5,0.5]]
